pages/react-context/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState } from "react";
+import {
+  initialThemeList,
+  ThemeList,
+} from "../../components/react-context/types";
+import ThemePicker from "../../components/react-context/ThemePicker";
+import ThemedForm from "../../components/react-context/ThemedForm";
+import { FormContext } from "../../components/react-context/FormContext";
+
+const ThemeCreator = () => {
+  const [themes, setThemes] = useState<ThemeList>(initialThemeList);
+  const [selectedTheme, setSelectedTheme] = useState("default");
+
+  return (
+    <div className="flex">
+      <section className="h-fit p-3 m-3 bg-slate-300 rounded">
+        <ThemePicker
+          onChangeHandler={(e) => setSelectedTheme(e.target.value)}
```

Comment:
This means your `useState` must reside in the same component as you create `ThemePIcker`.

Suppose you more components wrapping before you use `ThemePicker`.

```
const ThemeCreator = ({isWizard}:{isWizard: boolean}) =>  isWizard ? <ThemeWizardCreator/> : <ThemeAdvancedForm/>;

const ThemeAdvancedForm = () => {
   const [themes, setThemes] = useState<ThemeList>(initialThemeList);
   // ...
   return ...
      <ThemePicker onChangeHandler={...}/>
};
```

In this approach, your `ThemePicker` must the at the same level as the `useState`, this means the state cannot be shared with `ThemeWizardCreator`.

Then if you pull `useState` up one level, you'll have to pass `setThemes` to `ThemeAdvancedForm` through props.....
But `ThemeAdvancedForm` doesn't really need `setThemes`, it just passes it down to `ThemePicker`....so now you need to keep passing `setThemes` down to the component that actually uses it, which is also not good.